File: core/src/org/rexcellentgames/burningknight/entity/item/weapon/gun/BackGun.java
package org.rexcellentgames.burningknight.entity.item.weapon.gun;

import org.rexcellentgames.burningknight.assets.Locale;
import org.rexcellentgames.burningknight.util.Random;
import org.rexcellentgames.burningknight.util.geometry.Point;

public class BackGun extends GunA {
	{
		sprite = "item-back_gun";
		damage = 6;
		setAccuracy(2);
		name = Locale.get("back_gun");
		description = Locale.get("back_gun_desc");
	}

	@Override
	protected void sendBullets() {
		Point aim = this.owner.getAim();
		float a = (float) (this.owner.getAngleTo(aim.x, aim.y) + Math.PI);

		this.sendBullet((float) (a +
			Math.toRadians(Random.newFloat(-this.getAccuracy(), this.getAccuracy())))
		);
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/weapon/gun/BackGun.kt
package org.rexcellentgames.burningknight.entity.item.weapon.gun

import org.rexcellentgames.burningknight.util.Random

class BackGun : Gun() {
  init {
    damage = 10
    accuracy = 2f
    origin.x = 4f
    origin.y = 2f
    hole.x = 8f
    hole.y = 12f
    accuracy = 0f
    sprite = "item-back_gun"
  }

  override fun sendBullets() {
    val aim = this.owner.aim
    val a = (this.owner.getAngleTo(aim.x, aim.y) + Math.PI).toFloat()

    this.owner.knockback.x -= (Math.cos(a.toDouble()) * 60f).toFloat()
    this.owner.knockback.y -= (Math.sin(a.toDouble()) * 60f).toFloat()

    this.sendBullet((a + Math.toRadians(Random.newFloat(-this.accuracy, this.accuracy).toDouble())).toFloat())
  }
}